### DISCUSSION TITLE
Issues using custom emotion instance in webextension breaking underlying page that also uses emotion

### DISCUSSION THREAD

**Discussion by samuelmburu** [emotion-js/emotion#2267] [Q&A]
HI, I am working on a browser extension, but adding the `emotion` import in my content scripts on a page that also uses emotion as their JSS solution renders their page unusable. I presume the issue has to deal with colliding emotion caches. I implemented the code from the [createCache documentation](https://emotion.sh/docs/@emotion/cache#options) a, but the issue is still present. Any help will be super helpful 🤞.

My code is as follows:

```jss
// ./emotion-instance.js
import createEmotion from '@emotion/css/create-instance';

export const {
    flush,
    hydrate,
    cx,
    merge,
    getRegisteredStyles,
    injectGlobal,
    keyframes,
    css,
    sheet,
    cache,
} = createEmotion({
    key: 'cb',
    prepend: true
});
```

```jsx
// ../shared/withCustomEmotion
import { CacheProvider } from '@emotion/react';

import * as customCache from './emotion-instance';

function Wrapper() {
    // I don't render any visible content
    return <CacheProvider value={customCache}>pop</CacheProvider>
}

export { Wrapper };
```

```js
// contentScript.js
import browser from 'webextension-polyfill';
import React from 'react';

// ... some more ✨ import statements

import { Wrapper } from '../shared/withCustomEmotion'; 
// 👆 The moment I add this line, the underlying pages goes 💥
```

I've been banging my head over this for a while and wonder what I'm doing wrong or if I didn't get  👀 the docs closely enough.

I really appreciate any help you can provide.
Thanks!

**Comment by Andarist** [comment]
Is the underlying page using Emotion 10 or 11?

**Reply by samuelmburu** [reply]
One seems to be using what seems to be like Emotion 10 because they have `@emotion/core` in their output bundles. On the other site, I can only tell they are using emotion from the `e.getAttribute("data-emotion-"+n)...` in their vendor bundle.

**Reply by Andarist** [reply]
Ok, I think I see the problem. We are moving all existing Emotion styles during initialization~:
https://github.com/emotion-js/emotion/blob/418daad9f7ac0eac88f206e3c4aee4e7aca7deb4/packages/cache/src/index.js#L55-L65

Was this a problem when you were using Emotion 10? Or is your webextension new and you have started with Emotion 11 right away?

**Reply by samuelmburu** [reply]
The extension is new, and I started using v11 right away
